google-cloud-storage/src/main/java/com/google/cloud/storage/Chunker.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import com.google.cloud.storage.Crc32cValue.Crc32cLengthKnown;
+import com.google.protobuf.ByteString;
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+// TODO: Better name, externally we're treating a chunk to mean a set of messages for which
+//   they all must commit before proceeding.
```

Comment:
Clarify what this class does; IIUC, this is chunking data for gRPC writes of 2MiB

Reply:
added some javadocs to the class, and one of the two methods (since they're overloads). I've also renamed the class from "Chunker" to "ChunkSegmenter" to be more accurate with our shared terminology where a Chunk results in multiple messages.